#44821: include rti-connext-7.3.0 in base.yaml

lobolanja opened the pull request:
If you're making a new release with bloom please use bloom to create the pull request automatically (except for the naming review request which must be made manually).
If you've already run the release bloom has a `--pull-request-only` option you can use.-->

**As I understand it, previous updates to the rti-connext-dds packages were handled by OSRF—we provided the binaries, and they added them to the ROS bootstrap repositories. @mjcarroll can you confirm that? **

<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

rti-connext-dds-7.3.0

## Package Upstream Source:

TODO (waiting clarification from OSRF on how to proceed)

## Purpose of using this:

This new package will bring new features, improved stability, and bug fixes compared to rti-connext-dds-6.0.1. We’ve also already had ROS 2 users requesting this update.

## Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - REQUIRED [TODO] 
- Ubuntu: https://packages.ubuntu.com/
  - REQUIRED [TODO] we are waiting for getting assistance on this
- Fedora: https://packages.fedoraproject.org/
  - not available
- Arch: https://www.archlinux.org/packages/
  - not available
- Gentoo: https://packages.gentoo.org/
  - not available
- macOS: https://formulae.brew.sh/
  - not available
- Alpine: https://pkgs.alpinelinux.org/packages
  - not available
- NixOS/nixpkgs: https://search.nixos.org/packages
  - not available
- openSUSE: https://software.opensuse.org/package/
  - not available
- rhel: https://rhel.pkgs.org/
  - not available

<!-- DOC_INDEX_TEMPLATE: add package to rosdistro for documentation indexing -->

<!--- Templated for adding a package to be indexed in a rosdistro: http://wiki.ros.org/rosdistro/Tutorials/Indexing%20Your%20ROS%20Repository%20for%20Documentation%20Generation -->

# Please Add This Package to be indexed in the rosdistro.

ROSDISTRO NAME

# The source is here:

http://sourcecode_url

# Checks
 - [ ] All packages have a declared license in the package.xml
 - [ ] This repository has a LICENSE file
 - [ ] This package is expected to build on the submitted rosdistro

lobolanja committed:
#44821: include rti-connext-7.3.0 in base.yaml

lobolanja committed:
44821: apply format to a comment

lobolanja committed:
44821: remove bionic since it is a EOL ubuntu version.

lobolanja committed:
44821: point to rti-connext-dds-7.3.0-ros since this ebian package contains a license file to be used out of the box by ros users

lobolanja committed:
#44821: Point to new package with '-ros' suffix to avoid collision with RTI evaluation packages

lobolanja commented:
@nuclearsandwich do we need to address this PR before RMW freeze?

mjcarroll commented:
One approval from me, I'll let @nuclearsandwich get the second.

lobolanja committed:
#44821: fix ubuntu version since it is only for noble for the moment.

nuclearsandwich commented on rosdep/base.yaml:
```diff
@@ -8469,11 +8469,16 @@ rti-connext-dds-5.3.1:
+    focal: [rti-connext-dds-6.0.1]
+    jammy: [rti-connext-dds-6.0.1]
+    noble: [rti-connext-dds-6.0.1]
+rti-connext-dds-7.3.0-ros:  # supported from ubuntu 24.04 noble
```
rosdep convention is to match the Debian package name but I actually considered matching the previous RTI package name in #44972 and considering the `-ros` an implementation detail.

I think the main thing it affects is needing to know which connext versions use a `ros`-suffixed package when specifying connext version in package.xml dependencies. Whether that's enough of a gripe to be worth defying convention is not up to me.

lobolanja replied:
@nuclearsandwich 

I made that change mostly by intuition, since I wasn’t entirely sure how to represent it.
However, I’d prefer to keep the `-ros` suffix as an implementation detail.

To do that, would it be enough to change both parts to the following?

```
rti-connext-dds-7.3.0:
  nixos: []
  ubuntu:
    '*': [rti-connext-dds-7.3.0]
    jammy: null
    focal: null
```

lobolanja replied:
check changes

nuclearsandwich commented on rosdep/base.yaml:
```diff
@@ -8469,11 +8469,16 @@ rti-connext-dds-5.3.1:
+    noble: [rti-connext-dds-6.0.1]
+rti-connext-dds-7.3.0-ros:  # supported from ubuntu 24.04 noble
+  nixos: []
+  ubuntu:
+    noble: [rti-connext-dds-7.3.0-ros]
```
In order to maintain forward compatibility we usually invert the definition here:
```suggestion
  ubuntu:
    '*': [rti-connext-dds-7.3.0-ros]
    jammy: null
    focal: null
```

This will allow us to use this package in 26.04 without requiring re-definition. The method by which we currently populate bootstrap repositories will insure this package is propagated to 26.04 when we first create it. However, if the Connext maintainers prefer explicitly defining which distros are supported for each rosdep definition I'm alright with that.

lobolanja replied:
@nuclearsandwich Just a quick question to avoid any issues—can you confirm whether I should use `rti-connext-dds-7.3.0` instead of `rti-connext-dds-7.3.0-ros` in [this PR](https://github.com/ros2/rmw_connextdds/pull/181) in `rti_connext_dds_cmake_module/package.xml`?

nuclearsandwich commented on rosdep/base.yaml:
```diff
@@ -8469,11 +8469,16 @@ rti-connext-dds-5.3.1:
   nixos: []
   ubuntu:
-    '*': [rti-connext-dds-6.0.1]
-    bionic: null
+    focal: [rti-connext-dds-6.0.1]
+    jammy: [rti-connext-dds-6.0.1]
+    noble: [rti-connext-dds-6.0.1]
```
I would actually suggest leaving the rti-connext-dds-6.0.1 definition as-is. The suite could actually be completely elided since bionic is no longer a checked / supported distro but it doesn't do much harm to leave it as it was.

lobolanja replied:
done!

lobolanja committed:
#44821: leaving the rti-connext-dds-6.0.1 definition as it was

lobolanja committed:
#44821: come back to rti-connext-dds-7.3.0 (rti-connext-dds-7.3.0-ros will be keep as a implementation detail)

mjcarroll commented on rosdep/base.yaml:
```diff
@@ -8474,6 +8474,12 @@ rti-connext-dds-6.0.1:
   ubuntu:
     '*': [rti-connext-dds-6.0.1]
     bionic: null
+rti-connext-dds-7.3.0:
```
This line should match what is in the downstream package.xml (and it does).

lobolanja replied:
Thanks! @mjcarroll

mjcarroll commented on rosdep/base.yaml:
```diff
@@ -8474,6 +8474,12 @@ rti-connext-dds-6.0.1:
+rti-connext-dds-7.3.0:
+  nixos: []
+  ubuntu:
+    '*': [rti-connext-dds-7.3.0]
```
This line should match the name of the package that is to be installed from debs (so should have the `-ros` suffix).

lobolanja replied:
@mjcarroll thanks for clarifying.

nuclearsandwich-ai commented on rosdep/base.yaml:
```diff
@@ -8474,6 +8474,12 @@ rti-connext-dds-6.0.1:
+rti-connext-dds-7.3.0:
+  nixos: []
+  ubuntu:
+    '*': [rti-connext-dds-7.3.0]
```
```suggestion
    '*': [rti-connext-dds-7.3.0-ros]
```

Apologies for creating ambiguity here @lobolanja. The package is keeping the `-ros` suffix. My comments about dropping it only applied to the rosdep key name.

It's probably best just to keep it everywhere and follow convention. Sorry for the confusion.

lobolanja replied:
@nuclearsandwich @mjcarroll   I think we can go with that approach—keeping the `-ros` suffix only for the package that’s installed via the .deb, and using `rti-connext-dds-7.3.0` as the key and in the package.xml.

At the end of the day, the real dependency is Connext 7.3.0, so the `-ros` suffix doesn’t really matter from a functionality standpoint, it is just to differentiate the package from RTI regular Debian packages.

lobolanja committed:
Update rosdep/base.yaml

use rti-connext-dds-7.3.0-ros since this is the debian package name is going to be installed.

Co-authored-by: Steven! Ragnarök <122943030+nuclearsandwich-ai@users.noreply.github.com>

lobolanja commented on rosdep/base.yaml:
```diff
@@ -8474,6 +8474,12 @@ rti-connext-dds-6.0.1:
   ubuntu:
     '*': [rti-connext-dds-6.0.1]
     bionic: null
+rti-connext-dds-7.3.0:
```
@nuclearsandwich @mjcarroll it is fine tho have this "warning" ?

nuclearsandwich committed:
Fix lexicographical sort.

nuclearsandwich commented on rosdep/base.yaml:
```diff
@@ -8474,6 +8474,12 @@ rti-connext-dds-6.0.1:
   ubuntu:
     '*': [rti-connext-dds-6.0.1]
     bionic: null
+rti-connext-dds-7.3.0:
```
Intentional. Discussed here: https://github.com/ros/rosdistro/pull/44918#discussion_r2032162264
